runtime/shared_common/CacheMap.cpp

```diff
@@ -3120,17 +3120,43 @@ SH_CacheMap::updateROMClassResource(J9VMThread* currentThread, const void* addre
 				break;
 			}
 			U_8* updateAddress = (U_8*)resourceDescriptor->unWrap(resourceWrapper) + updateAtOffset;
+			const ShcItem *itemInCache = resourceDescriptor->wrapperToItem(resourceWrapper);
+			ShcItem *tmpItem = NULL;
+			const ShcItem *itemToUse = itemInCache;
+			bool addResourceInTopLayer = false;
 			if (false == isAddressInCache((void*)updateAddress, data->length, false, true)) {
-				Trc_SHR_CM_updateROMClassResource_Exit7(currentThread, updateAddress, data->length);
-				result = J9SHR_RESOURCE_STORE_ERROR;
-				break;
+				/* We cannot overwrite the existing resource which is in a lower layer cache, instead we add a new resource. 
+				 * The old resources will be removed from the hashtable. */
+				Trc_SHR_Assert_True(isAddressInCache((void*)updateAddress, data->length, false, false));
+				tmpItem = (ShcItem*)j9mem_allocate_memory(itemInCache->dataLen, J9MEM_CATEGORY_CLASSES);
+				if (NULL == tmpItem) {
+					result = J9SHR_RESOURCE_STORE_ERROR;
+					break;
+				} else {
+					memcpy(tmpItem, itemInCache, itemInCache->dataLen);
+					addResourceInTopLayer = true;
+					itemToUse = tmpItem;
+				}
 			}
 
-			const ShcItem *itemInCache = resourceDescriptor->wrapperToItem(resourceWrapper);
 			if (false == isUDATA) {
-				resourceDescriptor->updateDataInCache(itemInCache, updateAtOffset, data);
+				resourceDescriptor->updateDataInCache(itemToUse, updateAtOffset, data);
```

Comment:
Shouldn't this only be done if addResourceInTopLayer is false?

Reply:
When `addResourceInTopLayer` is false, itemToUse points to the `ShcItem` in the shared cache, so we are directly updating the data at `updateAtOffset` in the shared cache. 
When addResourceInTopLayer is true. `itemToUse` points to the local copy `tmpItem`, and we are updating the local copy at `updateAtOffset`. This local copy is then added to the top layer cache.